platform-sdk/docs/proposals/template.md

```diff
@@ -0,0 +1,135 @@
+# Proposal Title Goes Here
+
+---
+
+## Summary
+
+Give a 1-3 sentence summary of the capability in the design proposal.
+
+| Metadata           | Entities                            | 
+|--------------------|-------------------------------------|
+| Designers          | John, Mary, alpha-team              |
+| Functional Impacts | Services, DevOps, Mirror Node, etc. |
+| Related Proposals  | Proposal-1, Proposal-2              |
+| HIPS               | HIP-1, HIP-2,                       |
+
+---
+
+## Purpose and Context
+
+Why is the design proposal necessary? Describe the background and purpose of the design proposal at a high level.
+
+### Dependencies, Interactions, and Implications
+
+What are the dependencies and pre-requisites of this proposal?  
+What other proposals, capabilities, and teams are impacted by this proposal?
+What does the proposal enable once it is completed?
+
+Remove this section if not applicable.
+
+### Requirements
+
+Describe the requirements and acceptance criteria that the design proposal must satisfy.
+
+### Design Decisions
+
+Describe the decisions made and the reasons why.
+
+#### Alternatives Considered
+
+Describe any alternatives considered and why they were not chosen.
+
+If possible, provide a table illustrating the options, evaluation criteria, and scores that factored into the decision.
+
+
+---
+
+## Changes
+
+### Architecture and/or Components
+
+Describe any new or modified components or architectural changes. This includes thread management changes, state
+changes, disk I/O changes, platform wiring changes, etc. Include diagrams of architecture changes.
+
+Remove this section if not applicable.
+
+### Module Organization and Repositories
+
+Describe any new or modified modules or repositories.
+
+Remove this section if not applicable.
+
+### Core Behaviors
+
+Describe any new or modified behavior. What are the new or modified algorithms and protocols? Include any diagrams that
+help explain the behavior.
+
+Remove this section if not applicable.
+
+### Public API
+
+Describe any public API changes or additions. Include stakeholders of the API.
+
+Examples of public API include:
+
+* Anything defined in protobuf
+* Any functional API that is available for use outside the module that provides it.
```

Comment:
Question: Do we need to create a proposal whenever that happens? More concrete: If I add api that could be used from outside of a module do I need to add a proposal in future?